src/oceanic/OceanicEventHandler.cpp

```diff
@@ -0,0 +1,241 @@
+#include "pch/stdafx.h"
+#include "oceanic/OceanicEventHandler.h"
+#include "curl/CurlInterface.h"
+#include "curl/CurlRequest.h"
+#include "curl/CurlResponse.h"
+#include "dialog/DialogManager.h"
+#include "task/TaskRunnerInterface.h"
+#include "datablock/DatablockFunctions.h"
+
+namespace UKControllerPlugin {
+    namespace Oceanic {
+
+        const std::string nattrakUrl = "https://nattrak.vatsim.net/pluginapi.php";
+
+        OceanicEventHandler::OceanicEventHandler(
+            Curl::CurlInterface& curl,
+            TaskManager::TaskRunnerInterface& taskRunner,
+            Dialog::DialogManager& dialogManager
+        ): curl(curl), taskRunner(taskRunner), dialogManager(dialogManager) {}
+
+        void OceanicEventHandler::TimedEventTrigger()
+        {
+            this->taskRunner.QueueAsynchronousTask([this]()
+            {
+                LogInfo("Updating oceanic clearance data from Nattrak");
+                Curl::CurlRequest apiUpdateRequest(nattrakUrl, Curl::CurlRequest::METHOD_GET);
+                Curl::CurlResponse apiUpdateResponse = this->curl.MakeCurlRequest(apiUpdateRequest);
+
+                if (apiUpdateResponse.IsCurlError() || !apiUpdateResponse.StatusOk()) {
+                    LogWarning("Unable to retrieve oceanic clearances from Nattrak.");
+                    return;
+                }
+
+                nlohmann::json clearanceData;
+                try {
+                    clearanceData = nlohmann::json::parse(apiUpdateResponse.GetResponse());
+                } catch (nlohmann::json::exception exception) {
+                    LogWarning("Unable to decode oceanic clearances from Nattrak, JSON parse failed.");
+                    return;
+                }
+
+                if (!clearanceData.is_array()) {
+                    LogWarning("Unable to decode oceanic clearances from Nattrak, JSON is not array.");
+                    return;
+                }
+
+                // Loop the clearances and update local data
+                this->clearances.clear();
+                for (
+                    auto clearance = clearanceData.cbegin();
+                    clearance != clearanceData.cend();
+                    ++clearance
+                ) {
+                    if (!NattrakClearanceValid(*clearance)) {
+                        LogWarning("Invalid clearance received from Nattrak");
+                        continue;
+                    }
+
+                    clearances.insert(
```

Comment:
```suggestion
                    this->clearances.insert(
```
Just so it's a bit clearer with similar variables in the function.

Reply:
Done!